Support 'embeds' parameter to discord API

dogboydog opened the pull request:
Hi,

Support [embeds](https://birdie0.github.io/discord-webhooks-guide/structure/embeds.html) for discord notifications. Probably best for people to try to generate them in a previous step and set them there with ::set-env or ::set-output

Related #41 
Let me know any changes you would like.

Ilshidur commented:
Thank you for your PR ! I currently don't have a lot of time to walk through your code. I think I'll be able to take a look at it next week.

dogboydog commented:
Sounds good

On Thu, Aug 13, 2020, 9:16 AM Nicolas Coutin <notifications@github.com>
wrote:

> Thank you for your PR ! I currently don't have a lot of time to walk
> through your code. I think I'll be able to take a look at it next week.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ilshidur/action-discord/pull/42#issuecomment-673472121>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACLWDQZSY3P6KIEBHOVQQO3SAPRUFANCNFSM4P3YVMJQ>
> .
>

Ilshidur commented on entrypoint.js:
```diff
@@ -31,18 +31,28 @@ _.templateSettings.interpolate = /{{([\s\S]+?)}}/g;
   url = process.env.DISCORD_WEBHOOK;
   payload = JSON.stringify({
     content: message,
+    ...process.env.DISCORD_EMBEDS && { embeds: embedsObject},
```
```suggestion
    ...process.env.DISCORD_EMBEDS && { embeds: embedsObject },
```

Ilshidur reviewed (changes requested):
Ok, I finally had time to inspect your PR.

I'm pretty happy with the result. I added some docs and fixed some "linting" issues. Thing is, I did not install a linter yet in this repo... :smile: 

If you're OK with my changes request, I'll merge it and immediately cut a release featuring your changes.

dogboydog commented:
> 
> 
> Ok, I finally had time to inspect your PR.
> 
> I really like it. Also, I added some docs and fixed some "linting" issues. Thing is, I did not install a linter yet in this repo... 😄
> 
> If you're OK with my changes request, I'll merge it and immediately cut a release featuring your changes.

Sounds good to me! I committed your changes. Thanks for taking a look. After the merge, I might delete my fork. Once I ended up doing more customization logic for my discord notifications, it made sense to implement my own custom client in Python.  But I did test these changes which allowed `embeds` to be sent, so hopefully someone out there will get some use out of it.

Ilshidur reviewed (approved):
LGMT. Thanks again for this PR. I hope people will like it.

Let's go for the release.
